src/Microsoft.Performance.Toolkit.Engine/EngineExecutionResults.cs

```diff
@@ -215,5 +225,228 @@ public bool TryQueryOutput<T>(DataOutputPath dataOutputPath, out T data)
+                {
+                    return processorIsDataAvail;
+                }
+            }
+            catch (Exception inner)
+            {
+                throw new TableException(tableDescriptor, inner);
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        ///     Builds a table with the given <paramref name="tableDescriptor"/>.
+        /// </summary>
+        /// <param name="tableDescriptor">
+        ///     The table to build.
+        /// </param>
+        /// <returns>
+        ///     The built table.
+        /// </returns>
+        /// <exception cref="TableException">
+        ///     A table cannot be built for the given <paramref name="tableDescriptor"/>.
+        /// </exception>
+        public ITableResult BuildTable(TableDescriptor tableDescriptor)
+        {
+            var tableBuilder = new TableBuilder();
+
+            try
+            {
+                if (FuncOnRepositoryIfContained(tableDescriptor, (reference, tableRetrieval) => { reference.BuildTableAction(tableBuilder, tableRetrieval); return tableBuilder; }, out ITableResult repoTableResult))
+                {
+                    return repoTableResult;
+                }
+
+                if (FuncOnProcessorIfContained(tableDescriptor, (processor) => { processor.BuildTable(tableDescriptor, tableBuilder); return tableBuilder; }, out ITableResult processorTableResult))
+                {
+                    return processorTableResult;
+                }
+            }
+            catch (Exception inner)
+            {
+                throw new TableException(tableDescriptor, inner);
+            }
+
+            throw new TableException(tableDescriptor);
+        }
+
+        /// <summary>
+        ///     Attempts to build a table with the given <paramref name="tableDescriptor"/>.
+        /// </summary>
+        /// <param name="tableDescriptor">
+        ///     The table to build.
+        /// </param>
+        /// <param name="filledTableBulder">
+        ///     The built table, if any.
+        /// </param>
+        /// <returns>
+        ///     <c>true</c> if the table for the given <paramref name="tableDescriptor"/> was built;
+        ///     <c>false</c> otherwise.
+        /// </returns>
+        public bool TryBuildTable(TableDescriptor tableDescriptor, out ITableResult filledTableBulder)
+        {
+            try
+            {
+                filledTableBulder = BuildTable(tableDescriptor);
+                return true;
+            }
+            catch (TableException)
+            {
+                filledTableBulder = null;
+                return false;
+            }            
+        }
+
+        // Will perform the Func on the Repository if the table is contained, else false.
+        private bool FuncOnRepositoryIfContained<TResult>(TableDescriptor tableDescriptor, Func<ITableExtensionReference, IDataExtensionRetrieval, TResult> func, out TResult result)
```

Comment:
>FuncOnRepositoryIfContained [](start = 21, length = 27)

My personal preference is to avoid types in names - so something like "ExecuteOnRepositoryIfContained"... but not a big deal with way. Feel free to ignore :)